src/sdk/pynni/nni/gridsearch_tuner/gridsearch_tuner.py

```diff
@@ -135,9 +138,31 @@ def update_search_space(self, search_space):
 
     def generate_parameters(self, parameter_id):
         self.count += 1
-        if self.count > len(self.expanded_search_space)-1:
-            raise nni.NoMoreTrialError('no more parameters now.')
-        return self.expanded_search_space[self.count]
+        while (self.count <= len(self.expanded_search_space)-1):
+            _params_tuple = tuple(sorted(self.expanded_search_space[self.count].items()))
+            if _params_tuple in self.feed_data:
+                self.count += 1
+            else:
+                return self.expanded_search_space[self.count]
+        raise nni.NoMoreTrialError('no more parameters now.')
 
     def receive_trial_result(self, parameter_id, parameters, value):
         pass
+
+    def feed_tuning_data(self, data):
+        """Feed additional data for tuning
+
+        Parameters
+        ----------
+        data:
+            a list of dictionarys, each of which has at least two keys, 'parameter' and 'value'
+        """
+        _completed_num = 0
+        for trial_info in data:
+            logger.info("Start to feed data, the current progrss number %s" %_completed_num)
+            _completed_num += 1
+            assert "parameter" in trial_info
+            _params = trial_info["parameter"]
+            _params_tuple = tuple(sorted(_params.items()))
```

Comment:
Nested dict should be considered

Reply:
Nested dict can also be successfully processed if the user is using the **same search space**.

Reply:
Then what's the meaning of `.items()`? It will only converted the first layer. If the value of the root dict is also of type dict, items() will return pairs like (key, another_dict)

Reply:
If you want to know the meaning of `.items()`, you can search `python3 .items()` in google/bing.
It seems that if the search spaces are exactly the same, the value of the "dict value"(including value) is also the same?
Can you give a specific example that will cause an error?

Reply:
I mean the meaning of using `.items()`, that is, the purpose of using `.items()`, not the usage of `items()`......
`{a:1, b:{c:2, d:3}}.items()` will become tuples like this: `(a, 1), (b, {c:2, d:3}}`... `{c:2, d:3}` is also a dict. Assuming that you'd like to hash a dict, then it will fail if the original dict is a nested dict.

Reply:
Fixed